web/src/app/shared/tree/tree.component.spec.ts

```diff
@@ -22,30 +26,117 @@ class TreeVendorComponent {
   @Input() state;
   @Input() nodes;
   treeModel = {
-    getActiveNode() { }
+    getActiveNode() {
+    }
   }
 }
+
 // tslint:enable:component-selector
 
 describe('TaskanaTreeComponent', () => {
   let component: TaskanaTreeComponent;
   let fixture: ComponentFixture<TaskanaTreeComponent>;
   let classificationCategoriesService;
-
+  let classificationsService;
+  let moveNodeEvent;
+  let dropEvent;
 
   beforeEach(done => {
     const configure = (testBed: TestBed) => {
       testBed.configureTestingModule({
         imports: [AngularSvgIconModule, HttpClientModule],
         declarations: [TreeVendorComponent],
-        providers: [TreeService, ClassificationCategoriesService]
+        providers: [TreeService, ClassificationCategoriesService, ClassificationsService]
 
       })
     };
     configureTests(configure).then(testBed => {
       fixture = testBed.createComponent(TaskanaTreeComponent);
       classificationCategoriesService = testBed.get(ClassificationCategoriesService);
       spyOn(classificationCategoriesService, 'getCategoryIcon').and.returnValue(new Pair('assets/icons/categories/external.svg'));
+      classificationsService = TestBed.get(ClassificationsService);
+      spyOn(classificationsService, 'getClassifications').and.returnValue(new ClassificationResource(
+        {
+          'classificationSummaryResourceList': new Array<Classification>(
+            new Classification('id1', 'key1', 'MANUAL', 'type', 'DOMAIN_A', 'classification1', '',
+              1, 'service', new Links({'href': 'someurl'})),
+            new Classification('id2', 'key2', 'MANUAL', 'type', 'DOMAIN_A', 'classification2', 'id1',
+              1, 'service', new Links({'href': 'someurl'})),
+            new Classification('id3', 'key3', 'MANUAL', 'type', 'DOMAIN_A', 'classification3', 'id1',
+              3, 'service', new Links({'href': 'someurl'})),
+            new Classification('id4', 'key4', 'MANUAL', 'type', 'DOMAIN_A', 'classification4', '',
+              5, 'service', new Links({'href': 'someurl'})))
+        }, new Links({'href': 'someurl'})
+      ));
+      spyOn(classificationsService, 'putClassification').and.callFake(function (url, classification) {
+        return classification;
+      });
+      spyOn(classificationsService, 'getClassification').and.returnValue(new ClassificationDefinition('id1', 'key1', '',
+        '', 'MANUAL', 'DOMAIN_A', 'TASK', true, '019-04-10T10:23:34.985Z', '2019-04-10T10:23:34.985Z',
+        'classification1', 'description', 1, 'level', '', '', '', '', '', ''
+        , '', '', '', new LinksClassification({href: ''}, '', '', {href: ''}, {href: ''}, {href: ''})));
+      moveNodeEvent = {
```

Comment:
Remove unused properties or use the event itself creating it via html

Reply:
done